### PR TITLE
[Process] allow writing "prepared" command line.

### PR DESCRIPTION
**Pull request by Simperfit**
This documents the re-pushed prepared command line with a new style of writing it that does not break anything.

https://github.com/symfony/symfony/pull/32126

### CODE REVIEW

File: components/process.rst [hunk 1]
```diff
@@ -350,6 +350,18 @@ instead::
     );
     $process->run();
 
+Using the prepared command lines
```
Comment (OskarStark) [inline components/process.rst]: „Using a Prepared Command Line“

File: components/process.rst [hunk 2]
```diff
@@ -350,6 +350,18 @@ instead::
...
+--------------------------------
+
+This component also provides a way to use the process command with prepared using the double brackets notations:
+
+    use Symfony\Component\Process\Process;
+
+  $process = Process::fromShellCommandline('echo "$name"');
+  $process->run(null, array('name' => 'elsa'));
```
Comment (OskarStark) [inline components/process.rst]: Please use short array syntax

File: components/process.rst [hunk 3]
```diff
@@ -350,6 +350,18 @@ instead::
...
+
+Which means that you can use placeholder in order to have a process that can be changed only with the values and without changing the php code. The component will not escape the characters, you are responsible of doing  so.
```
Comment (OskarStark) [inline components/process.rst]: 2 spaces before the last word

Comment (OskarStark) [reply components/process.rst]: I would put the whole text parts before the Code example and not talk again about „the component“

File: components/process.rst [hunk 4]
```diff
@@ -350,6 +350,19 @@ instead::
     );
     $process->run();
 
+Using a Prepared Command Line
+--------------------------------
```
Comment (fabpot) [inline components/process.rst]: should the same length as the title itself to be valid

File: components/process.rst [hunk 5]
```diff
@@ -350,6 +350,19 @@ instead::
...
+
+This component also provides a way to use the process command with prepared using the double brackets notations:
```
Comment (fabpot) [inline components/process.rst]: missing word after `with prepared`

Comment (fabpot) [reply components/process.rst]: double bracket notation

File: components/process.rst [hunk 6]
```diff
@@ -350,6 +350,19 @@ instead::
...
+Which means that you can use placeholder in order to have a process that can be changed only with the values and without changing the php code.
```
Comment (fabpot) [inline components/process.rst]: use a placeholder?

Comment (fabpot) [reply components/process.rst]: PHP

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by OskarStark** [comment]
@Simperfit do you find some time to finish this PR? 😃

**Comment by Simperfit** [comment]
Ill do it asap ;)

**Comment by OskarStark** [comment]
Thanks for your work on this new feature!

**Comment by Simperfit** [comment]
Thanks @OskarStark

File: components/process.rst (hunk 5 above)
Comment (xabbuh) [inline components/process.rst]: What does "double bracket notation" mean?

**Pull request closed without merging**